pkg/component/observability/monitoring/blackboxexporter/blackboxexporter.go

```diff
@@ -89,11 +100,31 @@ func (b *blackboxExporter) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	return managedresources.CreateForShoot(ctx, b.client, b.namespace, ManagedResourceName, managedresources.LabelValueGardener, false, data)
+	if b.values.ClusterType == component.ClusterTypeSeed {
+		return managedresources.CreateForSeedWithLabels(ctx, b.client, b.namespace, b.managedResourceName(), false, map[string]string{v1beta1constants.LabelCareConditionType: v1beta1constants.ObservabilityComponentsHealthy}, data)
+	}
+	return managedresources.CreateForShoot(ctx, b.client, b.namespace, b.managedResourceName(), managedresources.LabelValueGardener, false, data)
 }
 
 func (b *blackboxExporter) Destroy(ctx context.Context) error {
-	return managedresources.DeleteForShoot(ctx, b.client, b.namespace, ManagedResourceName)
+	if b.values.ClusterType == component.ClusterTypeSeed {
+		return managedresources.DeleteForSeed(ctx, b.client, b.namespace, b.managedResourceName())
+	}
+	return managedresources.DeleteForShoot(ctx, b.client, b.namespace, b.managedResourceName())
+}
+
+func (b *blackboxExporter) managedResourceName() string {
+	if b.values.ClusterType == component.ClusterTypeSeed {
+		return "blackbox-exporter"
+	}
+	return "shoot-core-blackbox-exporter"
+}
+
+func (b *blackboxExporter) namespaceName() string {
+	if b.values.ClusterType == component.ClusterTypeSeed {
+		return b.namespace
+	}
+	return metav1.NamespaceSystem
 }
```

Comment:
I find this slightly confusing. There is `b.namespace` and `b.namespaceName()`. It seems like `b.namespace` is the namespace of the managed resource while `b.namespaceName()` is the namespace of the runtime components. Would it therefore make sense to rename the function to something like `runtimeNamespace()`?
WDYT?